browser/resources/settings/brave_privacy_page/brave_tor_subpage.js

```diff
@@ -0,0 +1,181 @@
+ * brave's Tor settings.
+ */
+class SettingsBraveTorPageElement extends SettingBraveTorPageElementBase {
+  static get is() {
+    return 'settings-brave-tor-subpage'
+  }
+
+  static get template() {
+    return html`{__html_template__}`
+  }
+
+  static get properties() {
+    return {
+      loadedConfig_: Object,
+      
+      useBridges_: Number,
+
+      builtInBridgesTypes_: {
+        type: Array,
+        value: [
+          { name: "snowflake", value: 0 },
+          { name: "obfs4", value: 1 },
+          { name: "meek-azure", value: 2 }
+        ]
+      },
+
+      builtinBridges_: Number,
+      requestedBridges_: Array,
+      providedBridges_: Array,
+
+      isUsingBridges_: {
+        type: Boolean,
+        notify: true,
+        value: false
+      },
+
+      showRequestBridgesDialog_: Boolean,
+    }
+  }
+
+  browserProxy_ = BraveTorBrowserProxyImpl.getInstance()
+
+  ready() {
+    super.ready()
+    this.browserProxy_.getBridgesConfig().then((config) => {
+      this.loadedConfig_ = config
+      this.UpdateUseBridges_(config.use_bridges)
+      this.builtinBridges_ = config.use_builtin_bridges
+      this.requestedBridges_ = config.requested_bridges
+      this.providedBridges_ = config.provided_bridges
+
+      this.$.builtInBridgesType.value = this.builtinBridges_
+
+      switch (this.useBridges_) {
+        case Usage.NOT_USED:
+        case Usage.USE_BUILT_IN:
+          this.$.brigdesGroup.selected = 'useBuiltIn'
+          break
+        case Usage.USE_REQUESTED:
+          this.$.brigdesGroup.selected = 'useRequestedBridges'
+          break
+        case Usage.USE_PROVIDED:
+          this.$.brigdesGroup.selected = 'useProvidedBridges'
+          break
+      }
+
+      this.$.requestedBridges.value = this.requestedBridges_.join('\n')
+      this.$.providedBridges.value = this.providedBridges_.join('\n')
+    })
+
+    this.$.useBuiltIn.expanded = true
+    this.$.useRequestedBridges.expanded = true
+    this.$.useProvidedBridges.expanded = true
+
+    this.$.providedBridges.oninput = (event) => {
+      this.providedBridges_ = this.onProvidedBridgesChange_(event)
+      this.handleChanged_()
+    }
+  }
+
+  getCurrentConfig_() {
+    const bridgesConfig = {
+      use_bridges: this.useBridges_,
+      use_builtin_bridges: Number.parseInt(this.builtinBridges_, 10),
+      requested_bridges: this.requestedBridges_,
+      provided_bridges: this.providedBridges_,
+    }
+    return bridgesConfig
+  }
+
+  UpdateUseBridges_(value) {
+    this.useBridges_ = value
+    this.isUsingBridges_ = value != Usage.NOT_USED
+  }
+
+  onUseBridgesChange_() {
+    if (!this.$.useBridges.checked) {
+      this.UpdateUseBridges_(Usage.NOT_USED)
+    } else {
+      this.onUsageSelect_()
+    }
+    this.handleChanged_()
+  }
+
+  onUsageSelect_() {
+    switch (this.$.brigdesGroup.selected) {
+      case 'useBuiltIn':
+        this.UpdateUseBridges_(Usage.USE_BUILT_IN)
+        this.OnBuiltInBridgesSelect_()
```

Comment:
A handler (`OnBuiltInBridgesSelect_`) that is tied to a UI element (`builtInBridgesType`) event should probably only be called by that event, otherwise the flow can get confusing. Instead try to set a property and use observers.

This particular function (`onUsageSelect_`) also can be more minimal - just set the relevant properties that the user has changed from the UI, and let the rest of the class properties and observers handle side-effects.